File: shared/runtime/state_registry.py
"""运行态state_registry。"""

from __future__ import annotations

from typing import Any

from shared.config.script_config import get_config
from shared.paths.script_paths import runtime_file_path
from shared.runtime.json_store import read_json, write_json


def create_state_registry(config: dict[str, Any] | None = None) -> dict[str, Any]:
    """创建统一的运行态文件注册表。"""

    loaded_config = config or get_config()
    runtime_files = loaded_config.get("storage", {}).get("runtime_files", {})

    def resolve(logical_key: str, fallback_name: str):
        configured = str(runtime_files.get(logical_key) or "").strip()
        return runtime_file_path(configured or fallback_name)

    def read(logical_key: str, fallback_name: str, fallback_value: Any):
        return read_json(resolve(logical_key, fallback_name), fallback_value)

    def write(logical_key: str, fallback_name: str, value: Any):
        return write_json(resolve(logical_key, fallback_name), value)

    return {
        "resolve": resolve,
        "read": read,
        "write": write,
    }

File: shared/runtime/state_registry.js
"use strict";

const { getConfig } = require("../config/node_config");
const { runtimeFilePath } = require("../paths/node_paths");
const { readJson, writeJson } = require("./json_store");

/**
 * 运行态state_registry。
 * 统一管理运行态 JSON 文件的路径、读取和写入。
 */
function createStateRegistry(options = {}) {
  const config = options.config || getConfig();
  const runtimeFiles = (config?.storage?.runtime_files && typeof config.storage.runtime_files === "object")
    ? config.storage.runtime_files
    : {};

  function resolve(logicalKey, fallbackName) {
    const configured = String(runtimeFiles?.[logicalKey] || "").trim();
    return runtimeFilePath(configured || fallbackName);
  }

  function read(logicalKey, fallbackName, fallbackValue) {
    return readJson(resolve(logicalKey, fallbackName), fallbackValue);
  }

  function write(logicalKey, fallbackName, value) {
    return writeJson(resolve(logicalKey, fallbackName), value);
  }

  return {
    resolve,
    read,
    write,
  };
}

module.exports = {
  createStateRegistry,
};


